great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py

```diff
@@ -187,91 +365,91 @@ def _build_parameters(
 """
             )
-        if all(
-            [
-                np.issubdtype(type(metric_value), np.integer)
-                for metric_value in metric_values
-            ]
-        ):
-            round_decimals = 0
 
-        # Obtain truncate_distribution directive from rule state (i.e., variables and parameters); from instance variable otherwise.
+        statistic_calculator: NumericMetricRangeMultiBatchStatisticCalculator = (
+            NumericMetricRangeMultiBatchStatisticCalculator(
+                batch_ids=batch_ids_for_metrics_calculations,
+                validator=validator_for_metrics_calculations,
+                metric_name=self._metric_name,
+                metric_domain_kwargs=metric_domain_kwargs,
+                metric_value_kwargs=metric_value_kwargs,
+                nan_raises_exception=nan_raises_exception,
+            )
+        )
+
+        metric_values: Union[
+            np.ndarray,
+            List[Union[int, np.int32, np.int64, float, np.float32, np.float64]],
+        ] = statistic_calculator.generate_distribution_sample(
+            randomized_data_point_identifiers=batch_ids_for_metrics_calculations
+        )
+        metric_value: Union[int, np.int32, np.int64, float, np.float32, np.float64]
+
         truncate_distribution: Dict[
             str, Union[Optional[int], Optional[float]]
-        ] = get_parameter_value_and_validate_return_type(
+        ] = self._get_truncate_distribution_using_heuristics(
+            metric_values=metric_values,
             domain=domain,
-            parameter_reference=self._truncate_distribution,
-            expected_return_type=dict,
             variables=variables,
             parameters=parameters,
         )
-        distribution_boundary: Optional[Union[int, float]]
-        if not all(
-            [
-                (
-                    distribution_boundary is None
-                    or is_numeric(value=distribution_boundary)
-                )
-                for distribution_boundary in truncate_distribution.values()
-            ]
-        ):
-            raise ge_exceptions.ProfilerExecutionError(
-                message=f"""The directive "truncate_distribution" for {self.__class__.__name__} must specify the
-[lower_bound, upper_bound] closed interval, where either boundary is a numeric value (or None).
-"""
-            )
+        lower_bound: Optional[Union[int, float]] = truncate_distribution.get(
+            "lower_bound"
+        )
+        upper_bound: Optional[Union[int, float]] = truncate_distribution.get(
+            "upper_bound"
+        )
 
-        metric_values = np.array(metric_values, dtype=np.float64)
+        round_decimals: int = self._get_round_decimals_using_heuristics(
+            metric_values=metric_values,
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
```

Comment:
Should `_get_round_decimals_using_heuristics` only be used if `round_decimals` is not passed in as a parameter to `__init__()`?

Reply:
@anthonyburdi There are two things going on here.  First, the answer is "negative" -- the heuristics are engaged exactly when the `round_decimals` is omitted; furthermore, they can override it.  This is because we do not have a mechanism to specify the directives at the domain level when the domain builder emits multiple domains.  Second, we need to improve on this situation and come up with a way to specify such directives as `round_decimals` at the individual domain level, even when the domain_builder emits multiple domains.  For example, we need to be able to communicate such domain expertise to the domain builder as "this is money representing bridge tolls, hence cannot be negative" -- until we build this, the heuristics make the outcome a little bit more user friendly (e.g., no negative table row counts).  Thanks!

Reply:
Replying to the two items:
1. That makes sense, we are in agreement though my initial question was not worded clearly. However the default value for `round_decimals` I think should be `None` (see my other comment).
2. I think we already have this functionality via specifying different rules. For example if you want different rounding based on different domains, you would specify two (or more) distinct rules with distinct domain builders that get the appropriate domains and then use different `round_decimals` directives in their parameter builders.

Reply:
@anthonyburdi Yes, we have it if we use separate domain rules, but we do not have this in the "one rule handles many" type of domain rules, such as the simple semantic column domain rule.  I am not sure what the best solution is, because in the situation with many columns, it might be too laborious and error prone to micromanage the rule as it finds columns.

Reply:
This is a great question for Design Review. 

Just my 2c - that complexity (matching Domains to which ParameterBuilders should be applied to them) is the same complexity whether it happens via two separate DomainBuilders or in some kind of filter in the ParameterBuilder (e.g. only apply this config for the ParameterBuilder to this subset of the Domains and this other config to other Domains). My previous comment was motivated by the fact that we have purpose built DomainBuilders for filtering Domains, rather than also build and maintain "Domain filters" for ParameterBuilders.

Reply:
@anthonyburdi I agree.  Both @talagluck  and I have just recognized this morning that we are still missing a simplification step in our architecture that runs through the rules, one domain at a time -- and I added a "TODO" remark in the column style domain builders to this effect.  Thanks!

Reply:
This was accomplished via `JIRA-305` and `JIRA-309` -- and now the architecture is much cleaner and easier to interpret and work with (in terms of writing understandable `YML` configuration files and improved code maintenance).